### PR TITLE
gsoc: Add Appveyor

### PR DESCRIPTION
**Pull request by satwikkansal**
Adds Appveyor configuration files for CI on windows.

Related to https://github.com/coala/coala-quickstart/issues/22

### CODE REVIEW

File: coala_quickstart/generation/FileGlobs.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def get_project_files(log_printer, printer, project_dir):
     printer.print()
 
     escaped_project_dir = glob_escape(project_dir)
+    print("escaped", escaped_project_dir)
```
Comment (hemangsk) [inline coala_quickstart/generation/FileGlobs.py]: Hey, think its an unintentional debug print?

**Comment by satwikkansal** [comment]
Yeah, it's still wip. Forgot to put the label :p

File: coala_quickstart/generation/FileGlobs.py (hunk 1 above)
Comment (satwikkansal) [inline coala_quickstart/generation/FileGlobs.py]: yeah :P

**Comment by jayvdb** [comment]
@satwikkansal, I dont understand why you have added a Circle CI commit here, and are changing `.ci/deps.sh`.  That script isnt used by the AppVeyor build, and it is mostly a CircleCI thing.

Changing something wrt Circle should be done in a separate PR please .

**Comment by satwikkansal** [comment]
@jayvdb Actually there was nothing wrong with circle. It's just that I had to change `python3` to `python` in [tests.sh](https://github.com/coala/coala-quickstart/blob/master/.ci/tests.sh#L120) because appveyor doesn't recognize `python3`.

Also, simply replacing `python3` by `python` in "tests.sh" made the Circle CI to fail because then it began using python 2.7 which coala doesn't supports. 

So I'm trying to figure out a way to make Circle CI use python 3 for the `python` command (currently the output of the line `python --version` in `deps.sh` for Circle CI is `2.7.something`.

Not sure if this is the right way to proceed, will like your inputs on this :sweat_smile:

**Comment by jayvdb** [comment]
> because appveyor doesn't recognize `python3`.

That is a significant problem that should be fixed (and solution replicated to other appveyor yml).  You may fix the fix 'upstream' at https://github.com/ogrisel/python-appveyor-demo , or you should also submit a bug report upstream if the problem exists there too.

**Comment by jayvdb** [comment]
If this is up for review, please post Appveyor build link

**Comment by satwikkansal** [comment]
Okay, I think it's up for review now.

I added the Travis CI commit, just to make sure all of the three CIs are working correctly (will remove it once everything is green)

Here's the Appveyor build link: https://ci.appveyor.com/project/satwikkansal/coala-quickstart/build/job/du2ng882wpjo3u5k

Travis build link: https://travis-ci.org/satwikkansal/coala-quickstart/builds/241219899

**Comment by satwikkansal** [comment]
Also, I will submit a bug report upstream regarding `python3` in Appveyor, but I adapting to the `python` behavior similar to other coala repositories would be a better option (https://github.com/coala/coala-quickstart/pull/98/commits/abeaf62161ceda718f74a94450c704c52b6889a0)

File: tests/generation/FileGlobs.py [hunk 2]
```diff
@@ -100,15 +100,21 @@ def test_get_project_files_gitignore(self):
...
-            self.assertIn(os.path.join(os.getcwd(), "src", "file.c"), res)
-            self.assertIn(os.path.join(os.getcwd(), "root.c"), res)
-            self.assertIn(os.path.join(os.getcwd(), "ignore_dir", "src.c"), res)
-            self.assertIn(os.path.join(os.getcwd(), "ignore_dir", "src.js"), res)
+
+            paths = [
+                os.path.join(os.getcwd(), "src", "file.c"),
+                os.path.join(os.getcwd(), "root.c"),
+                os.path.join(os.getcwd(), "ignore_dir", "src.c"),
+                os.path.join(os.getcwd(), "ignore_dir", "src.js")
```
Comment (jayvdb) [inline tests/generation/FileGlobs.py]: add trailing comma so a new item can be added to the list without modifying this line.

File: tests/generation/FileGlobs.py [hunk 3]
```diff
@@ -100,15 +100,21 @@ def test_get_project_files_gitignore(self):
 
     def test_get_project_files_ci_mode(self):
         orig_cwd = os.getcwd()
-        os.chdir(os.path.dirname(os.path.realpath(__file__)) +
```
Comment (jayvdb) [inline tests/generation/FileGlobs.py]: This should be a separate commit, and that commit should add a rule to .coafile so that this cant re-occur.
(Ideally a separate PR, as this one is already full)

**Comment by jayvdb** [comment]
> Also, I will submit a bug report upstream regarding python3 issue in Appveyor

Have you found or raised the bug upstream in https://github.com/ogrisel/python-appveyor-demo ?

File: .ci/tests.sh [hunk 4]
```diff
@@ -9,4 +9,4 @@ if [ "$system_os" == "LINUX" ] ; then
   args+=('--doctest-modules')
 fi
 
-python3 -m pytest "${args[@]}"
\ No newline at end of file
+python -m pytest $*
```
Comment (jayvdb) [inline .ci/tests.sh]: better fix : https://github.com/coala/coala-quickstart/pull/112

**Comment by satwikkansal** [comment]
> Have you found or raised the bug upstream in https://github.com/ogrisel/python-appveyor-demo ?

Actually I came to know that the `python3` doesn't work in windows because there's no `.exe` or `.cmd` or `.bat` file corresponding to `python3`. The command works only after changing the `.exe` name or probably when using some virtual environment.

File: tests/generation/FileGlobs.py (hunk 3 above)
Comment (satwikkansal) [inline tests/generation/FileGlobs.py]: Yes, that should be better.

File: .ci/tests.sh (hunk 4 above)
Comment (satwikkansal) [inline .ci/tests.sh]: Agreed, we can merge this PR after #112 is merged.

**Comment by jayvdb** [comment]
I've raised an issue upstream; I'll let you find it.  I dont want to cross-link the issue to this PR, as it is just a distraction.

**Comment by satwikkansal** [comment]
Moved https://github.com/coala/coala-quickstart/commit/a8e7e398db2c0b20e3c0ef7459dbeeca9a087f1a out of this PR as it is not specific to Appveyor.

Build link: https://ci.appveyor.com/project/satwikkansal/coala-quickstart/build/job/2jyy3ui8rq4mwa8l

File: .ci/appveyor.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+branches:
+  except:
+    - /^sils\/.*/
+
+install:
+  # Prepend newly installed Python to the PATH of this build (this cannot be
+  # done from inside the powershell script as it would require to restart
+  # the parent CMD process).
+  - set PATH=%PYTHON%;%PYTHON%\Scripts;%PATH%;C:\Program Files\Git\mingw64\bin
```
Comment (jayvdb) [inline .ci/appveyor.yml]: I doubt that `C:\Program Files\Git\mingw64\bin` is needed here.  If it is, the reason should be explained in a comment.

**Comment by jayvdb** [comment]
The commit message "Project.py: Change inline documentation" could be improved to be "Project.py: Remove OS specific doctest"

File: .ci/appveyor.yml (hunk 5 above)
Comment (satwikkansal) [inline .ci/appveyor.yml]: Yes, not required as there's no command related to git.

File: .ci/appveyor.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+install:
+  # Prepend newly installed Python to the PATH of this build (this cannot be
+  # done from inside the powershell script as it would require to restart
+  # the parent CMD process).
+  - set PATH=%PYTHON%;%PYTHON%\Scripts;%PATH%;
+
+  # Check that we have the expected version and architecture for Python
+  - "python --version"
+  - "python -c \"import struct; print(struct.calcsize('P') * 8)\""
```
Comment (satwikkansal) [inline .ci/appveyor.yml]: I copied this line from coala/coala but not sure if it is required here or not. I guess this differentiates b/w 32 bit and 64 bit architecture, but doesn't coala support both of them?

Comment (jayvdb) [inline .ci/appveyor.yml]: That part is fine.
https://github.com/coala/coala-quickstart/pull/98/files#diff-b19d2d01a06d5ca549e5616cddf67d5bR11 shows the 32 bit test environment, and four lines beneath it is the 64 bit test environment

**Comment by jayvdb** [comment]
reack f9bd418 e34a334 d25f3ca

**Comment by jayvdb** [comment]
@rultor merge
